src/components/side_panel/table_panel/table_panel.ts

```diff
@@ -83,14 +91,9 @@ export class TablePanel extends Component<Props, SpreadsheetChildEnv> {
     }
   }
 
-  onChangeNumberOfHeaders(ev: Event) {
-    const input = ev.target as HTMLInputElement;
-    const numberOfHeaders = parseInt(input.value);
-    const result = this.updateNumberOfHeaders(numberOfHeaders);
-
-    if (!result.isSuccessful) {
-      input.value = this.props.table.config.numberOfHeaders.toString();
-    }
+  onChangeNumberOfHeaders(value: string) {
+    const numberOfHeaders = parseInt(value);
```

Comment:
`NumberInput` could do the parsing itself ? Unless, we can't have something generic because some inputs accept floats and others only integers ?